includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php

```diff
@@ -0,0 +1,917 @@
+					'description' => __( 'Content of the sitewide notice.', 'buddypress' ),
+					'type'        => 'object',
+					'required'    => true,
+					'arg_options' => array(
+						'sanitize_callback' => null,
+						'validate_callback' => null,
+					),
+					'properties'  => array(
+						'raw'      => array(
+							'description' => __( 'Content for the sitewide notice, as it exists in the database.', 'buddypress' ),
+							'type'        => 'string',
+							'context'     => array( 'edit' ),
+						),
+						'rendered' => array(
+							'description' => __( 'HTML content for the sitewide notice, transformed for display.', 'buddypress' ),
+							'type'        => 'string',
+							'context'     => array( 'view', 'edit' ),
+							'readonly'    => true,
+						),
+					),
+				),
+				'date'      => array(
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( "The date of the sitewide notice, in the site's timezone.", 'buddypress' ),
+					'readonly'    => true,
+					'type'        => 'string',
+					'format'      => 'date-time',
+				),
+				'is_active' => array(
+					'context'     => array( 'edit' ),
+					'description' => __( 'Whether this notice is active or not.', 'buddypress' ),
+					'readonly'    => true,
+					'type'        => 'boolean',
+				),
+			),
+		);
+
+		/**
+		 * Filters the notice schema.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param array $schema The endpoint schema.
+		 */
+		return apply_filters( 'bp_rest_sitewide_notices_schema', $this->add_additional_fields_schema( $schema ) );
+	}
+
+	/**
+	 * Get the query params for sitewide notices collections.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                       = parent::get_collection_params();
+		$params['context']['default'] = 'view';
+
+		/**
+		 * Filters the collection query params.
+		 *
+		 * @param array $params Query params.
+		 */
+		return apply_filters( 'bp_rest_sitewide_notices_collection_params', $params );
+	}
+
+	/**
+	 * Prepare a notice for creation or update.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return BP_Messages_Notice|WP_Error Object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+
+		$schema          = $this->get_item_schema();
+		$notice_id       = isset( $request['id'] ) ? absint( $request['id'] ) : 0;
+		$prepared_item   = $this->get_notice_object( $notice_id );
+
+		// Notice ID.
+		if ( ! empty( $schema['properties']['id'] ) && ! empty( $prepared_item->id ) ) {
+			$prepared_item->id = $prepared_item->id;
+		}
+
+		// Notice subject.
+		if ( ! empty( $schema['properties']['subject'] ) && isset( $request['subject'] ) ) {
+			if ( is_string( $request['subject'] ) ) {
+				$prepared_item->subject = $request['subject'];
+			} elseif ( isset( $request['subject']['raw'] ) ) {
+				$prepared_item->subject = $request['subject']['raw'];
+			}
+		}
```

Comment:
I'd do like this:

```suggestion
		$subject = $request->get_param( 'subject' );
		if ( ! empty( $schema['properties']['subject'] ) && $subject ) {
			if ( is_string( $subject  ) ) {
				$prepared_item->subject = $subject;
			} elseif ( isset( $subject['raw'] ) ) {
				$prepared_item->subject = $subject['raw'];
			}
		}
```

Same for the `message` and `is_active` props